clarifai/client/compute_cluster.py

```diff
@@ -0,0 +1,195 @@
+
+    Example:
+        >>> from clarifai.client.compute_cluster import ComputeCluster
+        >>> compute_cluster = ComputeCluster(compute_cluster_id="compute_cluster_id", user_id="user_id")
+        >>> nodepool = compute_cluster.create_nodepool(nodepool_id="nodepool_id", config_filepath="config.yml")
+    """
+    if not os.path.exists(config_filepath):
+      raise UserError(f"Nodepool config file not found at {config_filepath}")
+
+    nodepool_config = self._process_nodepool_config(config_filepath)
+
+    if 'id' in nodepool_config:
+      nodepool_id = nodepool_config['id']
+      nodepool_config.pop('id')
+
+    request = service_pb2.PostNodepoolsRequest(
+        user_app_id=self.user_app_id,
+        compute_cluster_id=self.id,
+        nodepools=[resources_pb2.Nodepool(id=nodepool_id, **nodepool_config)])
+    response = self._grpc_request(self.STUB.PostNodepools, request)
+    if response.status.code != status_code_pb2.SUCCESS:
+      raise Exception(response.status)
+    self.logger.info("\nNodepool created\n%s", response.status)
+
+    return Nodepool.from_auth_helper(self.auth_helper, nodepool_id=nodepool_id)
+
+  def nodepool(self, nodepool_id: str) -> Nodepool:
+    """Returns a Nodepool object for the existing nodepool ID.
+
+    Args:
+        nodepool_id (str): The nodepool ID for the nodepool to interact with.
+
+    Returns:
+        Nodepool: A Nodepool object for the existing nodepool ID.
+
+    Example:
+        >>> from clarifai.client.compute_cluster import ComputeCluster
+        >>> compute_cluster = ComputeCluster(compute_cluster_id="compute_cluster_id", user_id="user_id")
+        >>> nodepool = compute_cluster.nodepool(nodepool_id="nodepool_id")
+    """
+    request = service_pb2.GetNodepoolRequest(
+        user_app_id=self.user_app_id, compute_cluster_id=self.id, nodepool_id=nodepool_id)
+    response = self._grpc_request(self.STUB.GetNodepool, request)
+
+    if response.status.code != status_code_pb2.SUCCESS:
+      raise Exception(response.status)
+    dict_response = MessageToDict(response, preserving_proto_field_name=True)
+    kwargs = self.process_response_keys(dict_response[list(dict_response.keys())[1]],
+                                        list(dict_response.keys())[1])
+
+    return Nodepool.from_auth_helper(auth=self.auth_helper, **kwargs)
+
+  def delete_nodepools(self, nodepool_ids: List[str]) -> None:
+    """Deletes list of nodepools for the compute cluster.
+
+    Args:
+        nodepool_ids (List[str]): The nodepool IDs of the compute cluster to delete.
+
+    Example:
+        >>> from clarifai.client.compute_cluster import ComputeCluster
+        >>> compute_cluster = ComputeCluster(compute_cluster_id="compute_cluster_id", user_id="user_id")
+        >>> compute_cluster.delete_nodepools(nodepool_ids=["nodepool_id1", "nodepool_id2"])
+    """
+    request = service_pb2.DeleteNodepoolsRequest(
+        user_app_id=self.user_app_id, compute_cluster_id=self.id, ids=nodepool_ids)
```

Comment:
Does this endpoint supports passing `ids` as string? If the user wants to delete a single nodepool, can it support passing it as an string. If not can we add a check to assert if the `nodepool_ids` is an list.

Reply:
assertion added